<launch>
	<node name="joy_node" pkg="joy" type="joy_node" output="screen"/>

	<!--remap to="B01/mavros/vision_pose/pose" from="vrpn_client_node/B01/pose"/> 
	<include file="$(find vrpn_client_ros)/launch/sample.launch" /-->

	<group ns="B01">

		<!--node name="tf2pose" pkg="drone_pose" type="tf2pose" output="screen">
			<param name="tf_target_frame" value="odom"/>
			<param name="tf_source_frame" value="base_link"/>
			<param name="pose_frame_id" value="world"/>
			<remap from="pose_topic" to="mavros/vision_pose/pose"/>
		</node-->

		<node name="drone_pose_node" pkg="drone_pose" type="drone_pose_node" output="screen">
		
                        <rosparam param="x_bounds"> [-5, 5] </rosparam>
			<rosparam param="y_bounds"> [-5, 5] </rosparam>
			<rosparam param="z_bounds"> [-1, 2] </rosparam>
			
			<rosparam param="takeoff_position"> [0, 4.3, 0.5] </rosparam>
			
			<param name="mavros_state_topic" value="mavros/state"/>
			<param name="joy_topic" value="/joy"/>
			<param name="odometry_topic" value="odometry_sensor1/odometry"/>
			<param name="setpoints_topic" value="mavros/setpoint_position/local"/>
			<param name="flight_mode_service" value="flight_mode"/>
			<param name="trajectory_topic" value="trajectory"/>

			<param name="arm_button" type="int" value="2"/>
			<param name="disarm_button" type="int" value="3"/>

			<param name="flightMode1_button" type="int" value="0"/>
			<param name="flightMode2_button" type="int" value="1"/>

			<param name="x_axis" type="int" value="4"/>
                        <param name="y_axis" type="int" value="3"/>
			<param name="z_axis" type="int" value="1"/>
			<param name="yaw_axis" type="int" value="0"/>
			
		</node>
	
		<include file="$(find mavros)/launch/px4.launch">
		 <arg name="fcu_url" value="/dev/fcu"/>
		</include>

		<param name="mavros/vision_pose/tf/listen" type="bool" value="true"/>
                <param name="mavros/vision_pose/tf/frame_id" type="string" value="odom"/>
                <param name="mavros/vision_pose/tf/child_frame_id" type="string" value="base_link"/>
                <param name="mavros/vision_pose/tf/rate_limit" type="double" value="50"/>

	</group>
	
	
</launch>


<!-- Microsoft Xbox 360 Wired Controller for Linux -->
<!-- Buttons => 0:A, 1:B, 2:X, 3:Y, 4:LB, 5:RB, 6:back, 7:start, 8:power -->
<!-- Axes => 0:L/R Left stick, 1:U/D Left stick, 2:LT, 3:L/R Right stick, 4:U/D Right stick, 5:RT -->  

<!-- F170 Logitech for Linux -->
<!-- Buttons => -->
